src/System.Drawing.Common/tests/ColorTranslatorTests.cs

```diff
@@ -197,6 +197,40 @@ public void FromHtml_String_ReturnsExpected(string htmlColor, Color expected)
             }
         }
 
+        [Theory]
+        [InlineData("'", typeof(ArgumentException))]
+        [InlineData("'\"", typeof(ArgumentException))]
+        [InlineData("\"'", typeof(ArgumentException))]
+        [InlineData("#", typeof(ArgumentException))]
+        [InlineData("#G12", typeof(FormatException))]
+        [InlineData("#G12345", typeof(FormatException))]
+        [InlineData("  #G12  ", typeof(ArgumentException))]
+        [InlineData("  #G12345  ", typeof(ArgumentException))]
+        [InlineData("#FFFFFFFFF", typeof(ArgumentException))]
+        [InlineData("0x", typeof(ArgumentException))]
+        [InlineData("0xFFFFFFFFF", typeof(ArgumentException))]
+        [InlineData("0xG12", typeof(ArgumentException))]
+        [InlineData("&h", typeof(ArgumentException))]
+        [InlineData("&hG12", typeof(ArgumentException))]
+        [InlineData("1,2", typeof(ArgumentException))]
+        [InlineData("1,2,3,4,5", typeof(ArgumentException))]
+        [InlineData("-1,2,3", typeof(ArgumentException))]
+        [InlineData("256,2,3", typeof(ArgumentException))]
+        [InlineData("1,-1,3", typeof(ArgumentException))]
+        [InlineData("1,256,3", typeof(ArgumentException))]
+        [InlineData("1,2,-1", typeof(ArgumentException))]
+        [InlineData("1,2,256", typeof(ArgumentException))]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Difference in behavior, netfx throws Exception or ArgumentException")]
+        public void FromHtml_Invalid_Throws(string htmlColor, Type exception)
+        {
+            using (new ThreadCultureChange(CultureInfo.InvariantCulture))
+            {
+                Exception exceptionThrows = Assert.Throws(exception, () => ColorTranslator.FromHtml(htmlColor));
```

Comment:
This is overkill: use AssertExtensions.Throws<ArgumentException, Exception>(“htmlColor”, ...)

And then you don’t have to skip these tests for netfx

Reply:
AssertExtension accept only ArgumentException based exception

Reply:
Hmmm maybe there’s an overload that takes two types?

Reply:
on  #25526 we already discuss with @safern

Reply:
The reason why the extension method takes ArgumentException is because they also take a param name, but we could add an extension method that takes 2 generic parameters accepting Exception based objects as you suggested in the issue. 

Also there is a [ThrowsAny](https://github.com/dotnet/corefx/blob/master/src/CoreFx.Private.TestUtilities/src/System/AssertExtensions.cs#L119) which for this cases I'm not a fan because that one is specific for a method that can throw different exceptions in different scenarios. Here is just 2 different platforms and when reading the test can get confusing. So I would go for adding another extension method that takes 2 generic Exception based arguments and an action as a parameter.

Reply:
@hughbe @safern i understand, i'll work on it.

Reply:
This tests can also be updated to use the new extension method you added.